### ISSUE TITLE
why not call train_op directly?

### ISSUE THREAD

**Issue by zhaowwenzhong** [auroua/InsightFace_TF#84]
# 3.7 define the optimize method
    opt = tf.train.MomentumOptimizer(learning_rate=lr, momentum=args.momentum)
    # 3.8 get train op
    grads = opt.compute_gradients(total_loss)
    update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
    with tf.control_dependencies(update_ops):
        train_op = opt.apply_gradients(grads, global_step=global_step)
    # train_op = opt.minimize(total_loss, global_step=global_step)

what is difference between #3.8 and train_op??
why not call train_op directly?

**Comment by gouthamvgk** [comment]
Before updating the weights in every step the moving_mean and moving_variance of batchnorm layer has to be updated. This update operation is got by tf.GraphKeys.UPDATE_OPS and then run using the tf.control_dependencies which will execute it before running the context.
